code/game/atoms.dm

```diff
@@ -891,9 +891,11 @@
   * Not recommended to use, listen for the [COMSIG_ATOM_DIR_CHANGE] signal instead (sent by this proc)
   */
 /atom/proc/setDir(newdir)
+	var/old_dir = dir
 	SHOULD_CALL_PARENT(TRUE)
-	SEND_SIGNAL(src, COMSIG_ATOM_DIR_CHANGE, dir, newdir)
+	SEND_SIGNAL(src, COMSIG_ATOM_DIR_CHANGE, old_dir, newdir)
 	dir = newdir
+	SEND_SIGNAL(src, COMSIG_ATOM_AFTER_DIR_CHANGE, old_dir)
```

Comment:
Please don't add a new signal. Just adapt the `vehicle_moved` proc or make a new one. You already have the information on which the new direction will be there.

Ideally the signal only sends the `newdir` and not the `dir` at all, which can be checked accessing the source since it hasn't changed yet.